lnwallet/channel.go

```diff
@@ -4688,13 +4653,59 @@ func (lc *LightningChannel) computeView(view *HtlcView,
 	// channel constraints to the final commitment state. If any fee
 	// updates are found in the logs, the commitment fee rate should be
 	// changed, so we'll also set the feePerKw to this new value.
-	filteredHTLCView, err := lc.evaluateHTLCView(
+	filteredHTLCView, uncommitted, err := lc.evaluateHTLCView(
 		view, &ourBalance, &theirBalance, nextHeight, whoseCommitChain,
-		updateState,
 	)
 	if err != nil {
 		return 0, 0, 0, nil, err
 	}
+	if updateState {
```

Comment:
I'd suggest pulling all of the code in this `if` out into a `LightningChannel.applyStateUpdate` method that's called here and in the test (tried in this [patch](https://github.com/user-attachments/files/17297552/update_state_closure.patch) and it's a minor lift).


We can leave a comment in the test that they used to be one function, and that the coverage should (eventually) be migrated to `computeView` _if_ it's not already covered.

Reply:
I can do this. However, I'm slightly nervous about it because if I make it a method, the bar for being well-defined is much higher.

This block appears to locking in the commit height of uncommitted payment descriptors *and* updating the lifetime payment flow statistics of the channel. I can't think of a coherent way to describe the *conjunction* of those things. What do you think about extracting out a more well-formed version of `setHeights` and `recordFlow` separately and calling them back to back here. Those at least are easier to give solid definitions and names for.

I mostly want to avoid attaching a function `applyStateUpdate` to `LightingChannel` for something so niche and incidental. If my suspicions are correct, you're asking me to do this to consolidate and reduce duplication of code that is fundamentally identical, not because it needs to be a singular function, and not also because it really should be a method on LightningChannel. Is that right?

Reply:
> What do you think about extracting out a more well-formed version of setHeights and recordFlow separately and calling them back to back here. 

Happy with this! 

> If my suspicions are correct, you're asking me to do this to consolidate and reduce duplication of code that is fundamentally identical, not because it needs to be a singular function, and not also because it really should be a method on LightningChannel. Is that right?

Yes that's correct - don't like having the duplication just for the sake of not having to change the tests during this refactor.

Reply:
This change should be included in the most recent push. Lmk what you think.